Makefile.local

```diff
@@ -1,24 +1,145 @@
+# to a branch upstream, creates the PR, and shows the URL.
+pr: update commit-remote-api show-pr-url
+
+# Clones the upstream repository (at $(CALICO_REPOSITORY_NAME)) and
+# then pulls in changes from there to this repository. Does not commit
+# any changes.
+pull-upstream-changes:
 	# Clone a temporary copy of the Calico repo at the given version.
-	rm -rf /tmp/calico-api-mirror
-	mkdir -p /tmp/calico-api-mirror
-	git clone --depth 1 git@github.com:tigera/calico-private.git -b $(CALICO_VERSION) /tmp/calico-api-mirror
+	$(info Cloning git@github.com:$(CALICO_REPOSITORY_NAME).git into $(CALICO_TEMP_DIR))
+	rm -rf $(CALICO_TEMP_DIR)
+	mkdir -p $(CALICO_TEMP_DIR)
+	git clone --quiet --depth 1 git@github.com:$(CALICO_REPOSITORY_NAME).git -b $(GIT_BRANCH) $(CALICO_TEMP_DIR)
+
 	# Remove local files - we'll add them back from the Calico repo's contents.
 	rm -r pkg/ build/ examples/ hack/
+
 	# Add in files from the Calico repo.
-	cp -r /tmp/calico-api-mirror/api/. .
-	cp /tmp/calico-api-mirror/lib.Makefile .
-	cp /tmp/calico-api-mirror/metadata.mk .
-	# Some files, we want to keep the local versions of. 
-	# For example, README content is different between the two locations.
-	git checkout Makefile.local README.md
+	cp -r $(CALICO_TEMP_DIR)/api/. .
+	cp $(CALICO_TEMP_DIR)/lib.Makefile .
+	cp $(CALICO_TEMP_DIR)/metadata.mk .
+
+# Restores any files that we want to keep even if they're different
+# upstream. For example, `README.md` is different in this repository
+# than the equivalent file in the upstream repository.
+restore-local-files:
+	$(info Restoring locally-kept file(s): $(KEEP_LOCAL_FILES))
+	git checkout --quiet $(KEEP_LOCAL_FILES)
 
+# Validates whether or not the repository is dirty; errors if there are
+# local changes.
 check-dirty:
 	git diff --quiet || (echo "Repository has local changes" && exit 1)
+
+# Shows the URL of the upstream pull request (if it exists, or an error otherwise)
+show-pr-url:
+	$(info )
+	$(info ** PR can be viewed at $(shell \
+		gh pr list --head $(ORGANIZATION):$(PR_BRANCH) --json url --jq '.[0].url' | \
+		grep -q '.' || \
+		echo '[error: no PR found]' \
+		) **)
+	$(info )
+	true
+
+# Takes the changes that have been made so far and commits them, pushes
+# them to a remote branch, then creates a PR for them if there isn't one already.
+#
+# Note that this target does not depend on `restore-local-files`; this is to allow
+# for local changes to be made before being committed (e.g. if you wanted to also
+# update the README.md file in this PR). Be aware that any local changes which exist
+# will be committed here!
+commit-remote-api:
+	$(info Creating new branch $(PR_BRANCH), committing changes to it, and creating a PR for it)
+	echo "* Creating new branch"
+	git switch --quiet -C $(PR_BRANCH)
+	git add .
+	git reset -q Makefile.local
+	echo "* Committing changes to branch"
+	git commit --quiet --message="Automatic API update from $(CALICO_REPOSITORY_NAME) $(LOCAL_REPOSITORY_REVISION)"
```

Comment:
I think LOCAL_REPOSITORY_VERSION would be more useful if it showed the commit ID of the calico-private repo that this was taken from instead.

Reply:
I opted to remove the commit ID, because I realized it can be quite wrong; this functionality allows for creating or updating of PRs in the API repository, which means that what was the `LOCAL_REPOSITORY_VERSION` information could become outdated. That could end up being worse than not being present at all.

If we wanted to preserve this information, we could commit it to the repository in e.g. an `.upstream_git_version` file or something similar and overwrite it when we make an update, but I'm not sure if that's necessary or useful. For now I feel as though it's best to leave it out.